File: 교공 알고리즘 스터디/15주차/위장.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes)
{
    int answer = 1;
    unordered_map<string, int> clothes_m;

    for (auto cloth : clothes)
    {
        clothes_m[cloth[1]]++;
    }

    for (auto spy_cloth : clothes_m)
    {
        answer *= spy_cloth.second + 1;
    }

    return answer - 1;
}

File: 프로그래머스 고득점 kit/Java/위장.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> closet = new HashMap<>();

        for (String[] cloth : clothes) {
            closet.put(cloth[1], closet.getOrDefault(cloth[1], 0) + 1);
        }

        for (String cloth : closet.keySet()) {
            answer *= closet.get(cloth) + 1;
        }
        return answer - 1;
    }
}